[Refac] PlayerInteraction & Input 리팩토링_King

Klistas opened the pull request:
- Input에서 Pack과 Unpack하던 부분을 Input에서 Bool 값을 받아 Interaction에서 처리.
- AddListener만 모아 AddEventListener() 만들어 사용.
- 필요없는 LayerMask 제거

Klistas committed:
[Fix] PlayerInteraction & Input 리팩토링

- Input에서 Pack과 Unpack하던 부분을 Input에서 Bool 값을 받아 Interaction에서 처리.
- AddListener만 모아 AddEventListener() 만들어 사용.
- 필요없는 LayerMask 제거

csy-59 reviewed (commented):
Layer를 삭제하는 방향 보다는, Layer를 오히려 사용하고 tag 쪽은 해당 레이어의 세부 구별에 사용하는 것이 더 좋은 방법이지 않을까 싶습니다...! 
ex. Item(Layer) 중 bed(tag)

csy-59 reviewed (commented):
- Input 부분에서 bool 타입 변수 프로퍼티화가 되어 있지 않습니다...!
- IsPackButtonPressed = OVRInput.GetDown(OVRInput.RawButton.B); 로 간략화 가능합니다.

Klistas committed:
[Refac] Input, Interaction, Movement 리팩토링

- LayerMask를 사용. tag는 세부 아이템 처리에만 사용하기로.
- InteractiveLayer는 6으로 임의 설정.
- IsPackButtonPressed 프로퍼티 화 및 간략화.
- Movement 코드 리팩토링 및 로직 문제 해결

Klistas committed:
Merge branch 'main' into Player_Interaction-&-Input-Refactoring_King

csy-59 reviewed (approved):
확인했습니다...! 빠른 수정 감사드립니다!

strange-tiger reviewed (approved):
확인했습니다! 수고 많으셨습니다!
